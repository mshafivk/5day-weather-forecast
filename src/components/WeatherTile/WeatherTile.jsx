import React from 'react';
import PropTypes from 'prop-types';
import classes from './WeatherTile.less';

export const WeatherTile = ({ dateTime, temp, humidity, weather }) => (
  <div className={classes.container}>
    <div className={classes.row}>
      <div data-testid="date" className={classes.summary}>
        {dateTime}
      </div>
    </div>
    <div className={classes.row}>
      <div>
        <span>Temperature : </span>
        <span>{Math.round(temp - 273.15)}Â°C</span>
      </div>
      <div>
        <span>Humidity : </span>
        <span>{humidity}%</span>
      </div>
    </div>
    <div className={classes.row}>
      <div data-testid="summary" className={classes.summary}>
        {weather.length === 1 && weather[0].description}
      </div>
    </div>
  </div>
);

WeatherTile.propTypes = {
  dateTime: PropTypes.string,
  temp: PropTypes.number,
  humidity: PropTypes.number,
  weather: PropTypes.arrayOf(
    PropTypes.shape({ main: PropTypes.string, description: PropTypes.string }),
  ),
};
