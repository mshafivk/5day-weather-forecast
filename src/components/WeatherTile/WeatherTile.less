@import '../../common/styles/typography.less';
@import '../../common/styles/color.less';

.container {
  display: flex;
  flex-direction: column;
  background-color: @bg-color-secondary;
  color: @fg-color-secondary;

  .row {
    display: flex;
    flex-direction: row;

    div {
      .list-item();
      flex-grow: 1;
      border: 1px solid;
    }

    .summary {
      .list-item-strong();
      text-align: center;
    }
  }
}
