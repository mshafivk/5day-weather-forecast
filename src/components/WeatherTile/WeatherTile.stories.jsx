import React from 'react';
import {WeatherTile} from './WeatherTile';

export default {
  title: 'WeatherTile',
  component: WeatherTile,
};

const Tpl = (args) => <WeatherTile {...args} />;

export const WeatherTileDefault = Tpl.bind({});
WeatherTileDefault.args = {
  dateTime: '2020-08-04 18:00:00',
  temp: 234.23,
  humidity: 84,
  weather: [{main: 'Rain', description: 'light rain'}],
};

WeatherTileDefault.storyName = 'Default Tile';
