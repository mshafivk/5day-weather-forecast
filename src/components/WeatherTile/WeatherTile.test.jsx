import React from 'react';
import { WeatherTile } from './WeatherTile';
import { shallow } from 'enzyme';

describe('WeatherTile Component', () => {
  it('should render properly', () => {
    const props = {
      dateTime: '2020-08-04 18:00:00',
      temp: 234.23,
      humidity: 84,
      weather: [{ main: 'Rain', description: 'light rain' }],
    };
    const wrapper = shallow(<WeatherTile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('summary should be empty of weather object is passed as empty', () => {
    const props = {
      temp: 234.23,
      humidity: 84,
      weather: [],
    };
    const wrapper = shallow(<WeatherTile {...props} />);
    const summary = wrapper.find('[data-testid="summary"]');
    expect(summary).toHaveLength(1);
    expect(summary.text()).toBe('');
  });

  test('summary should show description passed from weather object', () => {
    const props = {
      temp: 234.23,
      humidity: 84,
      weather: [{ main: 'Rain', description: 'light rain' }],
    };
    const wrapper = shallow(<WeatherTile {...props} />);
    const summary = wrapper.find('[data-testid="summary"]');
    expect(summary).toHaveLength(1);
    expect(summary.text()).toBe('light rain');
  });
});
