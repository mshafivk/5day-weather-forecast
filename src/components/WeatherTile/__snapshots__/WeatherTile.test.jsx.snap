// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherTile Component should render properly 1`] = `
<div>
  <div>
    <div
      data-testid="date"
    >
      2020-08-04 18:00:00
    </div>
  </div>
  <div>
    <div>
      <span>
        Temperature : 
      </span>
      <span>
        234.23
      </span>
    </div>
    <div>
      <span>
        Humidity : 
      </span>
      <span>
        84
      </span>
    </div>
  </div>
  <div>
    <div
      data-testid="summary"
    >
      light rain
    </div>
  </div>
</div>
`;
